libs/data-service-generator/src/register-plugin.ts

```diff
@@ -36,17 +43,34 @@ async function* getPluginFuncGenerator(
   }
 }
 
+async function getPlugin(
+  packageName: string,
+  customPath: AlternativeImportFunction | undefined
+): Promise<any> {
+  if (!customPath) {
+    return await import(packageName);
+  }
+  const path = await customPath(packageName);
+  if (path) {
+    return await import(path);
+  }
+}
+
 /**
  * loop through all plugin list and set the plugin under each event
  */
 const getAllPlugins = async (
-  pluginList: PluginInstallation[]
+  pluginList: PluginInstallation[],
+  alternativeImport?: AlternativeImportFunction
 ): Promise<Events[]> => {
   if (!pluginList.length) return [];
 
   const pluginFuncsArr: Events[] = [];
 
-  for await (const pluginFunc of getPluginFuncGenerator(pluginList)) {
+  for await (const pluginFunc of getPluginFuncGenerator(
```

Comment:
you can either use a generator to make each download and yield result or wrap this loop with try catch. I would go with generator